router/router.go

```diff
@@ -5,21 +5,23 @@ import (
 	"time"
 
 	"github.com/gorilla/mux"
-	"github.com/raymundo/academy-go-q42021/controller"
-	"github.com/raymundo/academy-go-q42021/global"
+	"github.com/rayl17/academy-go-q42021/controller"
+	parameters "github.com/rayl17/academy-go-q42021/global"
 )
 
-func NewRouter() *http.Server {
+func NewRouter(c controller.ControllerInterface) *http.Server {
+
 	router := mux.NewRouter()
 
-	router.HandleFunc("/pokemon/{name}", controller.GetPokemonByName).Methods("GET")
-	router.HandleFunc("/pokemonid/{id}", controller.GetPokemonById).Methods("GET")
+	router.HandleFunc("/pokemon/{name}", c.GetPokemonByName).Methods("GET")
+	router.HandleFunc("/pokemonid/{id}", c.GetPokemonById).Methods("GET")
+	router.HandleFunc("/pokemon/{name}", c.PostPokemon).Methods("POST")
```

Comment:
Distinct routes should be separated for better readability.

Also, not sure if this is a best practice or not for REST API, but remember handling with a comment scenario.

- First, the base should be `pokemons` (in plural), because there might be an endpoint where I could get all of them and would be weird to call the `/pokemon` endpoint.
- Second, `id` would be the default search param, which also need to have the same `pokemons` base, and not a different one `pokemonid`
- Last, as the param after `pokemons` would be the `id` you should have an alternate route for other searching params, `name` as in your implementation. `pokemons/name/{name}` endpoint is easier to read and infer what is this about.


```suggestion
	router.HandleFunc("/pokemons/{id}", c.GetPokemonById).Methods("GET")
	
	router.HandleFunc("/pokemons/name/{name}", c.GetPokemonByName).Methods("GET")
	router.HandleFunc("/pokemons/name/{name}", c.PostPokemon).Methods("POST")
```